includes/abstracts/llms-abstract-query.php

```diff
@@ -0,0 +1,589 @@
+			'per_page'         => 10,
+			'search'           => '',
+			'sort'             => array(),
+			'suppress_filters' => false,
+			'no_found_rows'    => false,
+		);
+
+	}
+
+	/**
+	 * Determines the total number of found results for the given query and returns it.
+	 *
+	 * @since [version]
+	 *
+	 * @return int
+	 */
+	abstract protected function found_results();
+
+	/**
+	 * Retrieve a query variable with an optional fallback value when the value is not set.
+	 *
+	 * @since 3.8.0
+	 * @since [version] Moved from `LLMS_Database_Query` and updated to use the null coalesce operator.
+	 *
+	 * @param string $key     Variable key.
+	 * @param mixed  $default Default value.
+	 * @return mixed
+	 */
+	public function get( $key, $default = '' ) {
+		return $this->query_vars[ $key ] ?? $default;
+	}
+
+	/**
+	 * Get the final combined arguments used to generate the query.
+	 *
+	 * Retrieves the value of the protected `$arguments` variable.
+	 *
+	 * @since [version]
+	 *
+	 * @return array
+	 */
+	public function get_arguments() {
+		return $this->arguments;
+	}
+
+	/**
+	 * Get the default arguments used for the query.
+	 *
+	 * Retrieves the value of the protected `$arguments_default` variable.
+	 *
+	 * @since [version]
+	 *
+	 * @return array
+	 */
+	public function get_arguments_default() {
+		return $this->arguments_default;
+	}
+
+	/**
+	 * Get the original, uncleaned, arguments submitted to the query.
+	 *
+	 * Retrieves the value of the protected `$arguments_original` variable.
+	 *
+	 * @since [version]
+	 *
+	 * @return array
+	 */
+	public function get_arguments_original() {
+		return $this->arguments_original;
+	}
+
+	/**
+	 * Get the
+	 *
+	 * Retrieves the value of the protected `$query` variable.
+	 *
+	 * @since [version]
+	 *
+	 * @return mixed
+	 */
+	public function get_query() {
+		return $this->query;
+	}
+
+	/**
+	 * Retrieve a list of fields that are allowed to be used for result sorting.
+	 *
+	 * @since [version]
+	 *
+	 * @return string[]
+	 */
+	protected function get_allowed_sort_fields() {
+
+		$allowed_fields = $this->allowed_sort_fields;
+
+		if ( $this->get( 'suppress_filters' ) ) {
+			return $allowed_fields;
+		}
+
+		/**
+		 * Filters the allowed sort fields.
+		 *
+		 * The dynamic portion of this hook, `$this->type`, refers to the award type
+		 * for the query, either "achievements" or "certificates".
+		 *
+		 * @since [version]
+		 *
+		 * @param array $allowed_fields Default arguments.
+		 */
+		return apply_filters( "llms_{$this->id}_query_allowed_sort_fields", $allowed_fields );
+
+	}
+
+	/**
+	 * Retrieve default arguments for the query.
+	 *
+	 * @since 3.8.0
+	 * @since 4.5.1 Added new default arg `no_found_rows` set to false.
+	 * @since [version] Moved from `LLMS_Database_Query` abstract.
+	 *
+	 * @return array
+	 */
+	protected function get_default_args() {
+
+		if ( $this->get( 'suppress_filters' ) ) {
+			return $this->default_arguments();
+		}
+
+		/**
+		 * Filters the query default args.
+		 *
```

Comment:
Misses “The dynamic portion..."
```suggestion
		 * The dynamic part of the filter `$this->id` identifies the extending query.
		 *
```